src/main/java/org/openrewrite/java/migrate/lombok/LombokValueToRecord.java

```diff
@@ -0,0 +1,356 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.java.migrate.lombok;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.*;
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.AnnotationMatcher;
+import org.openrewrite.java.JavaIsoVisitor;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.RemoveAnnotationVisitor;
+import org.openrewrite.java.search.MaybeUsesImport;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.tree.*;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.Objects.requireNonNull;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class LombokValueToRecord extends ScanningRecipe<Map<String, Set<String>>> {
+
+    private static final Pattern LOMBOK_ANNOTATION_PATTERN = Pattern.compile("^lombok.*");
+
+    private static final String LOMBOK_VALUE_IMPORT = "lombok.Value";
+
+    private static final String LOMBOK_VALUE_ANNOTATION = "@lombok.Value";
+
+
+    @Option(displayName = "useExactToString",
+            description = "When set the `toString` format from Lombok is used in the migrated record.")
+    boolean useExactToString;
+
+    @JsonCreator
+    public LombokValueToRecord(final @JsonProperty("useExactToString") boolean useExactToString) {
+        this.useExactToString = useExactToString;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return String.format("Convert `%s` class to Record", LOMBOK_VALUE_ANNOTATION);
+    }
+
+    @Override
+    public String getDescription() {
+        return "Convert Lombok `@Value` annotated classes to standard Java Records.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("lombok");
+    }
+
+    @Override
+    public Map<String, Set<String>> getInitialValue(final ExecutionContext ctx) {
+        return new ConcurrentHashMap<>();
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getScanner(final Map<String, Set<String>> acc) {
+        final TreeVisitor<?, ExecutionContext> check = Preconditions.and(
+                new UsesJavaVersion<>(17),
+                new MaybeUsesImport<>(LOMBOK_VALUE_IMPORT)
```

Comment:
Rather than checking the import we can instead check for `UsesType`. Otherwise we might fail to pick up qualified use through `@lombok.Value`.